UnicornRender/video/source/Primitives.cpp

```diff
@@ -61,13 +65,76 @@ void Primitives::Cube(Mesh& mesh)
 void Primitives::Quad(Mesh& mesh)
 {
     std::vector<Vertex> vertices{{
-        {{-1.0f , -1.0f , 0.0f}, {1.0f , 0.0f}} ,
-        {{1.0f , -1.0f , 0.0f}, {0.0f , 0.0f}} ,
-        {{1.0f , 1.0f , 0.0f}, {0.0f , 1.0f}},
-        {{-1.0f , 1.0f , 0.0f}, {1.0f , 1.0f}},
+        {{-0.5f , -0.5f , 0.0f},{1.0f , 0.0f}} ,
+        {{0.5f , -0.5f , 0.0f},{0.0f , 0.0f}} ,
+        {{0.5f , 0.5f , 0.0f},{0.0f , 1.0f}},
+        {{-0.5f , 0.5f , 0.0f},{1.0f , 1.0f}},
     }};
 
     mesh.SetMeshData(vertices, {0, 1, 2, 2, 3, 0});
 }
+
+void Primitives::Sphere(Mesh& mesh, float radius, uint32_t rings, uint32_t sectors)
+{
+    if(radius < 0)
+    {
+        LOG_WARNING("Sphere radius less than 0! UV will be inverted!");
+    }
+    if(rings < 4 || sectors < 4)
+    {
+        LOG_WARNING("Rings or sectors are less than 4, sphere will not be generated!");
+        return;
+    }
+
+    std::vector<uint32_t> indices;
+    std::vector<Vertex> temp_vertices;
+
+    uint32_t vectorSize = rings * sectors;
+
+    if(vectorSize > temp_vertices.max_size())
```

Comment:
you're not checking against `numeric_limits<uint32_t>::max()` as it is mentioned in documentation